Libplanet/Net/Protocols/RoutingTable.cs

```diff
@@ -250,37 +301,79 @@ internal void AddPeer(BoundPeer peer, DateTimeOffset updated)
                 );
             }
 
-            _logger.Debug(
-                "Adding a peer {Peer} to peer {Address}'s routing table...",
-                peer,
-                _address);
-            BucketOf(peer).AddPeer(peer, updated);
+            if (_staticPeerStates.ContainsKey(peer.Address))
+            {
+                _logger.Verbose("The peer {Peer} to be added is already in static peer.", peer);
+
+                // This done because peer's other attribute except public key might be changed.
+                // (eg. public IP address, endpoint)
+                PeerState state = _staticPeerStates[peer.Address];
+                state.Peer = peer;
+                state.LastUpdated = updated;
+            }
+            else
+            {
+                _logger.Debug(
+                    "Adding a peer {Peer} to peer {Address}'s routing table...",
+                    peer,
+                    _address);
+                BucketOf(peer).AddPeer(peer, updated);
+            }
         }
 
         internal IEnumerable<BoundPeer> PeersToBroadcast(Address? except, int min = 10)
         {
-            List<BoundPeer> peers = NonEmptyBuckets
-                .Select(bucket => bucket.GetRandomPeer(except))
-                .Where(peer => !(peer is null)).ToList();
-            var count = peers.Count;
+            var random = new Random();
+            List<BoundPeer> candidates = new List<BoundPeer>();
+            for (int i = 0; i < TableSize; i++)
+            {
+                if (_buckets[i].IsEmpty() && !_staticPeerByBucketIndex.ContainsKey(i))
+                {
+                    continue;
+                }
+
+                var peers = new List<BoundPeer>();
+                if (!_buckets[i].IsEmpty())
+                {
+                    peers.AddRange(
+                        _buckets[i].Peers.Where(
+                            peer => except is null || !peer.Address.Equals(except.Value)));
+                }
+
+                if (_staticPeerByBucketIndex.ContainsKey(i))
+                {
+                    peers.AddRange(
+                        _staticPeerByBucketIndex[i].Where(
+                                address => except is null || !address.Equals(except.Value))
+                            .Select(address => _staticPeerStates[address].Peer));
+                }
+
+                if (peers.Any())
+                {
+                    candidates.Add(peers[random.Next(peers.Count)]);
+                }
+            }
```

Comment:
- This code seems to replace `KBucket.GetRandomPeer()`, right? if so, we can delete `KBucket.GetRandomPeer()` because it's used only here.
- But on the one hand, it would be better if we keep the previous implementation and add only static peer at last... 🤔